packages/block-library/src/search/style.scss

```diff
@@ -1,17 +1,54 @@
 .wp-block-search {
-	display: flex;
-	flex-wrap: wrap;
+	.wp-block-search__inside-wrapper {
+		display: flex;
+		flex: auto;
+		flex-wrap: nowrap;
+		max-width: 100%;
+	}
 
 	.wp-block-search__label {
 		width: 100%;
 	}
 
 	.wp-block-search__input {
 		flex-grow: 1;
-		max-width: 360px;
+		min-width: 3em;
+		border: 1px solid $dark-gray-200;
 	}
 
 	.wp-block-search__button {
 		margin-left: 10px;
+		word-break: normal;
+
+		svg {
+			min-width: 1.5em;
+			min-height: 1.5em;
+		}
+	}
+
+	&.wp-block-search__button-only {
+		.wp-block-search__button {
+			margin-left: 0;
+		}
+	}
+
+	&.wp-block-search__button-inside .wp-block-search__inside-wrapper {
+		padding: $grid-unit-05;
+		border: 1px solid $dark-gray-200;
+
+		.wp-block-search__input {
+			border-radius: 0;
+			border: none;
+			padding: 0 0 0 4px;
```

Comment:
Should this one be `0.25em` instead? :thinking:

Reply:
Oops, missed that. Fixing.